Changed default port

UpTide opened the pull request:
Changed the default port to be in the dynamic range, added a check to see… if requested port was out of range, and added an error message if user enters an out of range port.

UpTide committed:
Changed default port to be in the dynamic range, added a check to see if requested port was out of range, and added an error message if user enters an out of range port.

UpTide commented:
Let me know if you have any questions or if I should change anything please!

UpTide committed:
Removed comments to pass code style check.

Mailaender commented:
:+1: as according to https://en.wikipedia.org/wiki/List_of_TCP_and_UDP_port_numbers the previous 1234 is part of the registered section and already used. Plus this gets rid of magic numbers in the source code.

pchote commented on OpenRA.Game/Settings.cs:
```diff
@@ -39,7 +39,22 @@ public class ServerSettings
+				return this.listenPort;
+			}
+
+			set
```
Please move the validation logic into `ServerCreationLogic`. These settings classes are automatically synced with files on disk, so shouldn't contain any logic that isn't related to this purpose.

UpTide committed:
Moved port check logic out of settings.

UpTide commented:
I put the `new IPEndPoint()` inside the `Server` constructor to catch if the port number was too high. (It was unhandled before.)
I also utilized the existing error handler and removed the one I added earlier.

UpTide commented:
I think I might have pushed the standalone server start scripts too by mistake. I was testing them to make sure they still worked after my changes. Let me know if I need to take them out.

pchote commented on OpenRA.Game/Server/Server.cs:
```diff
@@ -121,10 +121,13 @@ public void EndGame()
 			Log.AddChannel("server", "server.log");
 
+			if (port < 49152 | port > 65534)
+				throw new System.Net.Sockets.SocketException(10022);
```
I'm all for hard-blocking ports outside this range from the in-game UI, but somewhere near 100% of our community hosted dedicated servers will also be caught by this.  Considering that we rely on the good-will of these hosts, forcing this on dedicated servers seems like a really bad idea.

Can we please relax this to a warning in the server log?

pchote commented on OpenRA.Mods.Common/Widgets/Logic/ServerCreationLogic.cs:
```diff
@@ -180,7 +180,7 @@ void CreateAndJoin()
 			int listenPort;
 			if (!Exts.TryParseIntegerInvariant(panel.Get<TextFieldWidget>("LISTEN_PORT").Text, out listenPort))
-				listenPort = 1234;
+				listenPort = Game.Settings.Server.ListenPort;
```
We can then add a check and specific "Port must be between 49151 and 65535" error dialog here.

LipkeGu replied:
we could put a warning and implement a method which is using 1234 as alternate port

reaperrr commented:
There haven't been any updates, so I'm closing this to clean up the PR queue and moving to Future milestone.
